File: Javascript/BrickWall.js
// Leetcode Problem - 554

var leastBricks = function(wall){
    const hashMap = new Map()

    for (let i =0;i<wall.length;i++){
        let widthSum = 0
        for (let j=0;j<wall[i].length-1;j++){
            widthSum += wall[i][j]
            hashMap.set(widthSum,(hashMap.get(widthSum) || 0) + 1)
        }
    }
    console.log(hashMap)
    let maxValue = 0
    for (const [key,value] of hashMap){
        if (value > maxValue){
            maxValue = value
        }
    }

    return (wall.length - maxValue)

}


const wall =[[1,2,2,1],
             [3,1,2],
             [1,3,2],
             [2,4],
             [3,1,2],
             [1,3,1,1]]
console.log(leastBricks(wall))

File: Python/BrickWall.py
## Leetcode Problem - 554

def leastBricks(wall):
    hashMap = {}

    for i in range(len(wall)):
        widthSum = 0
        for j in range(len(wall[i])-1):
            widthSum += wall[i][j]
            hashMap[widthSum] = hashMap.get(widthSum,0) + 1

    maxValue = 0

    print(hashMap)

    for key,value in hashMap.items():
        if value > maxValue:
            maxValue = value

    return len(wall) - maxValue


wall = [[1,2,2,1],
        [3,1,2],
        [1,3,2],
        [2,4],
        [3,1,2],
        [1,3,1,1]]
print(leastBricks(wall))